torch/csrc/api/src/optim/lbfgs.cpp

```diff
@@ -29,6 +33,10 @@ void LBFGS::add_grad(const torch::Tensor& step_size, const Tensor& update) {
   NoGradGuard guard;
   int64_t offset = 0;
   for (auto& parameter : parameters_) {
+    if (!parameter.grad().defined()) {
+      continue;
+    }
```

Comment:
I suspect that we don't need this change, because this function doesn't access `parameter.grad()` after the check.